dig.go

```diff
@@ -127,6 +145,14 @@ func Group(group string) ProvideOption {
 	})
 }
 
+// As is a ProvideOption that specifies that the struct produced by the constructor
+// implements the given interface
+func As(i interface{}) ProvideOption {
```

Comment:
We can also update this to `As(...interface{})`, too. That would simplify some of your test cases below.

Reply:
It would simplify the test cases, but introduce two ways of doing the same thing:

`dig.Provide(myfunc, dig.As(foo, bar, baz)`
`dig.Provide(myfunc, dig.As(foo), dig.As(bar), dig.As(baz)`

I think the Go way is to usually give you one way to do things (no while loops, etc) but I really don't have a strong preference here.

The only thing to consider is wether the `dig.As` *itself* needs options.

`dig.Provide(myfunc, dig.As(new(bytes.Buffer), VerySpecificAsOption(7))`